pkg/provisioner/ironic/clients/auth.go

```diff
@@ -0,0 +1,95 @@
+package clients
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// AuthType is the method of authenticating requests to the server
+type AuthType string
+
+const (
+	// NoAuth uses no authentication
+	NoAuth AuthType = "noauth"
+	// HTTPBasicAuth uses HTTP Basic Authentication
+	HTTPBasicAuth AuthType = "http_basic"
+)
+
+// AuthConfig contains data needed to configure authentication in the client
+type AuthConfig struct {
```

Comment:
Does this type need to be public? It doesn't look like its fields are used outside of this package.

Reply:
It's used in get-hardware-details.

Reply:
The type is, but not its fields, right? It could be an interface or the fields could be private.

Reply:
Oh yeah - in earlier commits it used the fields directly, but after 65f3c5fd86bc01d7376bf0fa81b4d3e0365693d3 you're right, they could be private.